client/src/pages/RegisterPage.js

```diff
@@ -1,43 +1,142 @@
-import React from "react";
-import NavBar from "../components/Navbar";
-import SelectDropDownComponent from "../components/Dropdown";
+import React, { useEffect, useState } from 'react';
+import Cookies from 'js-cookie';
+import NavBar from '../components/Navbar';
+import Dropdown from '../components/Dropdown';
+import DepartmentApi from '../api/DepartmentApi';
+import AuthApi from '../api/AuthApi';
+import RoleApi from '../api/RoleApi';
 
 const RegisterPage = () => {
+  const [departments, setDepartments] = useState([]);
+  const [roles, setRoles] = useState([]);
+  const [token, setToken] = useState('');
+  const [params, setParams] = useState({
+    fullname: '',
+    email: '',
+    department_id: '',
+    role: '',
+    avatar: `https://joeschmoe.io/api/v1/0`,
+  });
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    DepartmentApi.fetchDepartments().then((res) => {
+      setDepartments(res.data);
+    });
+    RoleApi.fetchRoles().then((res) => {
+      setRoles(res.data);
+    });
+    setToken(Cookies.get('token'));
+    Cookies.get('params') &&
+      setParams(JSON.parse(Cookies.get('params') || '{}'));
+  }, []);
+
+  const handleInputChange = (e) => {
```

Comment:
I think it would be nice to add ```setErrorMessage('')``` on this line so that the error message will disappear upon reinput of the user